src/tools/compiletest/src/main.rs

```diff
@@ -1,10 +1,16 @@
-use std::{env, sync::Arc};
+use std::{env, io::IsTerminal, sync::Arc};
 
 use compiletest::{common::Mode, log_config, parse_config, run_tests};
 
 fn main() {
     tracing_subscriber::fmt::init();
 
+    // colored checks stdout by default, but for some reason only stderr is a terminal.
+    // compiletest *does* print many things to stdout, but it doesn't really matter.
+    if std::io::stderr().is_terminal() && std::env::var("NO_COLOR") != Ok("1".to_owned()) {
```

Comment:
Suggestion: maybe just check if `NO_COLOR` is set? i.e. `std::env::var_os("NO_COLOR").is_some()`

Reply:
i went with `matches!(std::env::var("NO_COLOR").as_deref(), None | Some("0"))`, I think `NO_COLOR=0` should still enable it, it's the same behavior as `colored`